frontend/src/modals/delete-document/DeleteDocumentModal.tsx

```diff
@@ -32,17 +52,50 @@ const deleteDocumentMutation = gql`
   }
 `;
 
-const propTypes = {
-  documentName: PropTypes.string.isRequired,
-  documentId: PropTypes.string.isRequired,
-  onDocumentDeleted: PropTypes.func.isRequired,
-  onClose: PropTypes.func.isRequired,
-};
+const isErrorPayload = (payload: GQLDeleteDocumentPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+export const DeleteDocumentModal = ({
+  documentName,
+  documentId,
+  onDocumentDeleted,
+  onClose,
+}: DeleteDocumentModalProps) => {
+  const [{ value, context }, dispatch] = useMachine<DeleteDocumentModalContext, DeleteDocumentModalEvent>(
+    deleteDocumentModalMachine
+  );
+  const { toast, deleteDocumentModal } = value as SchemaValue;
+  const { message } = context;
+
+  const [performDocumentDeletion, { loading, data, error }] = useMutation<GQLDeleteDocumentMutationData>(
+    deleteDocumentMutation
+  );
+  useEffect(() => {
+    if (!loading) {
+      if (error) {
+        const showToastEvent: ShowToastEvent = {
+          type: 'SHOW_TOAST',
+          message: error.message,
+        };
+        dispatch(showToastEvent);
+      } else if (data) {
```

Comment:
You can have both errors and data so do not use:

```
if (error) {
} else if (data) {
}
```

but instead 

```
if (error) {
}
if (data) {
}
```

Reply:
Done